dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/restfulWS/client/security/LtpaHandler.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.restfulWS.client.security;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.ws.rs.ProcessingException;
+import jakarta.ws.rs.client.ClientRequestContext;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.security.web.WebSecurityHelper;
+
+public class LtpaHandler {
+    private static final TraceComponent tc = Tr.register(LtpaHandler.class);
+
+    public static void configClientLtpaHandler(ClientRequestContext crc) {
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "configClientLtpaHandler - About to get a LTPA authentication token");
+        }
+
+        // retrieve a ltpa cookie from the Subject in current thread
+        try {
+            // this interceptor must depend on the appSecurity feature to use WebSecurityHelper.getSSOCookieFromSSOToken()
+            Cookie ssoCookie = WebSecurityHelper.getSSOCookieFromSSOToken();
+            if (ssoCookie == null) {
+                return;
+            }
+            String cookieName = ssoCookie.getName();
+            String cookieValue = ssoCookie.getValue();
+            if (cookieValue != null && !cookieValue.isEmpty() && cookieName != null) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Retrieved a LTPA authentication token. About to set a request cookie: " + cookieName + "=" + cookieValue);
+                }
+                crc.getHeaders().putSingle("Cookie", ssoCookie.getName() + "=" + ssoCookie.getValue());
+            } else { // no user credential available
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "Cannot find a ltpa authentication token off of the thread, you may need enable feature appSecurity-2.0 or ssl-1.0");
```

Comment:
Since you updated to beans 4.0 in the beans.xml, it seems you are going with EE 10 only now with the PR and so this would be appSecurity-5.0.  Either leave things be EE9 for now for dependencies and switch to beans 3.0 in the beans.xml or move everything to EE10 in the buildpath and Java 11.  The easiest would just be updating to beans 3.0 for now.

Reply:
Can leave for now, but will need to determine which path, but can be done in a next PR.